src/python/pants/goal/run_tracker.py

```diff
@@ -32,6 +32,21 @@
 from pants.version import VERSION
 
 
+@dataclass
+class PantsDaemonStats:
+    """Tracks various stats about the daemon."""
+
+    scheduler_metrics: Dict[str, int] = field(default_factory=dict)
+
+    def set_scheduler_metrics(self, scheduler_metrics: Dict[str, int]) -> None:
+        self.scheduler_metrics = scheduler_metrics
+
+    def get_all(self) -> Dict[str, int]:
+        for key in ["target_root_size", "affected_targets_size"]:
+            self.scheduler_metrics.setdefault(key, 0)
```

Comment:
Is this stale?

Reply:
I am not sure, I am still trying to trace out all the code paths involved here. If it is stale I'll remove it in a followup commit to come shortly.

Reply:
I think we should kill this loop, if the metrics are not there, we shouldn't be forcing them into this dict.

Reply:
I am fine with throwing away this data class  also.
I think it is pointless...

we can just change the API on run tracker and be done with it.

Reply:
https://github.com/pantsbuild/pants/pull/11003